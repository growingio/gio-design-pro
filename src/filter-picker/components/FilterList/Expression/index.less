.mock-attribute-select {
  width: 160px;
  height: 20px;
  background: burlywood;
}

.expression-box {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 40px;
  padding-right: 16px;
  margin: 8px;
  .express-regular_select {
    display: flex;
    justify-content: flex-start;
    align-items: center;
  }
  .expression-icon {
    width: 24px;
    height: 24px;
    text-align: center;
    line-height: 24px;
    background-color: #dfe4ee;
    border-radius: 12px;
    margin-right: 8px;
  }

  .filter-condition_select {
    border-radius: 4px;
    position: relative;
    // background-color: #f1f2f8;
    box-sizing: border-box;
    max-width: 160px;
    // width: 122px;
    width: auto;
    min-width: 40px;
    height: 30px;
    display: inline-block;
    padding: 4px 8px;
    line-height: 22px;
    text-align: center;
    vertical-align: middle;
    font-size: 14px;
    color: #313e75;
    margin-left: 8px;

    &:hover {
      background-color: #f1f2f8;
    }
    .filter-condition_select-text {
      width: 100%;
      display: inline-block;
      padding-right: 20px;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
    }
  }
  // 旧版属性选择器样式
  .dimension-select-dropdown {
    // display: flex;
    // align-items: center;
    .ant-select-dropdown-menu {
      .ant-select-dropdown-menu-item {
        padding: 9px 8px 10px 35px;
      }

      .ant-select-dropdown-menu-item-group {
        .ant-select-dropdown-menu-item-group-title {
          color: #222f73;
          font-weight: normal;
          margin-left: 10px;
          line-height: inherit;

          .group-icon {
            display: inline-block;
            vertical-align: bottom;
            width: 20px;
            height: 20px;
            background-color: #e3e7fc;
            border-radius: 50%;
            text-align: center;

            svg {
              margin-right: 0 !important;
            }
          }
        }
      }
    }
  }
}
