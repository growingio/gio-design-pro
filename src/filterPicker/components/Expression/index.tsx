import React from 'react';

function Expression() {
  return <div>expresion</div>;
}

export default Expression;
