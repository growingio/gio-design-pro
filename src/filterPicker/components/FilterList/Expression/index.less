.mock-attribute-select {
  width: 160px;
  height: 20px;
  background: burlywood;
}

.expression-box {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 40px;
  padding-right: 16px;
  margin: 8px;
  .express-regular_select {
    display: flex;
    justify-content: flex-start;
    align-items: center;
  }
  .expression-icon {
    width: 24px;
    height: 24px;
    text-align: center;
    line-height: 24px;
    background-color: #dfe4ee;
    border-radius: 12px;
    margin-right: 8px;
  }

  .filter-condition_select {
    border-radius: 4px;
    background-color: #f1f2f8;
    width: 122px;
    height: 30px;
    display: inline-block;
    line-height: 30px;
    text-align: center;
    vertical-align: middle;
    font-size: 14px;
    color: #313e75;
  }
}
