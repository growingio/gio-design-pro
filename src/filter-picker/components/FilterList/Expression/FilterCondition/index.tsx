import React, { useMemo, useState } from 'react';
import { DownFilled } from '@gio-design/icons';
import { Dropdown, Tooltip } from '@gio-design/components';
import moment from 'moment';

import FilterAttrOverlay from './FilterAttrOverlay';
import { attributeValue, FilterValueType, StringValue, NumberValue, DateValue } from './interfaces';

interface FilterConditionProps {
  valueType: attributeValue;
  onSubmit: (v: FilterValueType) => void;
  onCancel: () => void;
  op: StringValue | NumberValue | DateValue;
  dimensionValueRequest?: (data: any) => Promise<any>;
  timeRange: string;
  measurements: any[];
  values: string[];
  exprKey: string;
}

interface operationMapType {
  string: { [key: string]: string };
  int: { [key: string]: string };
  date: { [key: string]: string };
  STRING: { [key: string]: string };
}

const operationMap: operationMapType = {
  string: {
    '=': '等于',
    '!=': '不等于',
    in: '在,范围内',
    'not in': '不在,范围内',
    like: '包含',
    'not like': '不包含',
    hasValue: '有值',
    noValue: '无值',
  },
  STRING: {
    '=': '等于',
    '!=': '不等于',
    in: '在,范围内',
    'not in': '不在,范围内',
    like: '包含',
    'not like': '不包含',
  },
  int: {
    '=': '等于',
    '!=': '不等于',
    '>': '大于',
    '>=': '大于等于',
    '<': '小于',
    '<=': '小于等于',
    between: '在,与,之间',
    'not between': '不在,与,之间',
    noValue: '无值',
    hasValue: '有值',
  },
  date: {
    '=': '等于',
    '!=': '不等于',
    '>': '在,天之后',
    '>=': '在,天之后包括当天',
    '<': '在,天之前',
    '<=': '在,天之前包括当天',
    between: '在,与,之间',
    'not between': '不在,与,之间',
    relativeCurrent: '相对现在',
    relativeBetween: '相对区间',
    hasValue: '有值',
  },
};

function FilterCondition(props: FilterConditionProps) {
  const {
    valueType = 'string',
    onSubmit,
    onCancel,
    op,
    dimensionValueRequest,
    timeRange,
    measurements,
    values,
    exprKey,
  } = props;
  const [visible, setVisible] = useState(false);
  // 对NumberAttrSelect，StringAttrSelect，DateAttrSelect返回的values值进行转换,生成属性选择框的属性规则展示文本
  const parseValuesToText = (type: attributeValue, operation: string, value: string[]): string => {
    const opMap = operationMap[type];
    if (value.length) {
      if (type === 'string' || type === 'STRING') {
        // 字符串类型
        switch (operation) {
          // 判断当op ： ”=“ 时，是否为无值状态，如果无值，返回’有值’
          case '=': {
            if (value[0] === ' ') {
              return '无值';
            }
            return opMap[operation] + value[0];
          }
          // 判断当op ： ”！=“ 时，是否为有值状态，如果有值，返回’无值‘
          case '!=': {
            if (value[0] === ' ') {
              return '有值';
            }
            return opMap[operation] + value[0];
          }

          case 'in':
          case 'not in': {
            const textList = opMap[operation].split(',');
            return textList[0] + value.join(',') + textList[1];
          }
          default:
            return opMap[operation] + value[0];
        }
      }
      if (type === 'int') {
        switch (operation) {
          // 判断当op ： ”！=“ 时，是否为有值状态，如果有值，返回’有值‘
          case '!=': {
            if (value[0] === ' ') {
              return '有值';
            }
            return opMap[operation] + value[0];
          }
          // 判断当op ： ”=“ 时，是否为无值状态，如果无值，返回’无值‘
          case '=': {
            if (value[0] === ' ') {
              return '无值';
            }
            return opMap[operation] + value[0];
          }
          case 'between':
          case 'not between': {
            const textList = opMap[operation].split(',');
            // 返回 ------'在(value1)与(value2)之间'
            return textList[0] + value[0] + textList[1] + value[1] + textList[2];
          }
          default:
            return opMap[operation] + value[0];
        }
      }
      if (type === 'date') {
        switch (operation) {
          case '!=': {
            if (value[0] === ' ') {
              return '有值';
            }
            return opMap[operation] + moment(value[0]).format('YYYY-MM-DD');
          }
          case '=': {
            if (value[0] === ' ') {
              return '无值';
            }
            return opMap[operation] + moment(value[0]).format('YYYY-MM-DD');
          }
          case '>':
          case '>=':
          case '<':
          case '<=': {
            const textList = opMap[operation].split(',');
            return textList[0] + moment(value[0]).format('YYYY-MM-DD') + textList[1];
          }
          // 判断，在。。。与。。。之间
          // 返回字符串 ---- ‘在（date1）与（data2）之间’
          case 'between':
          case 'not between': {
            const textList = opMap[operation].split(',');
            if (value[0].includes('abs')) {
              const abs = value?.[0].split(':')[1].split(',');
              return (
                textList[0] +
                moment(parseInt(abs[0], 10)).format('YYYY-MM-DD') +
                textList[1] +
                moment(parseInt(abs[1], 10)).format('YYYY-MM-DD') +
                textList[2]
              );
            }
            break;
          }
          case 'relativeTime': {
            const relativeTime = value[0]
              .split(':')[1]
              .split(',')
              .map((ele: string) => parseInt(ele, 10));
            if (relativeTime.length === 1) {
              const day = relativeTime[0];
              if (day < 0) {
                return `过去${Math.abs(day)}天前`;
              }
              return `未来${Math.abs(day)}天后`;
            }
            if (relativeTime.includes(0)) {
              if (relativeTime[0]) {
                if (relativeTime[0] < 0) {
                  return `过去${Math.abs(relativeTime[0])}天内`;
                }
                return `未来${Math.abs(relativeTime[0])}天内`;
              }
              if (relativeTime[1] < 0) {
                return `过去${Math.abs(relativeTime[1])}天内`;
              }
              return `未来${Math.abs(relativeTime[1])}天内`;
            }
            if (relativeTime[0] < 0) {
              return `过去${Math.abs(relativeTime[0])}-${Math.abs(relativeTime[1])}天内`;
            }
            return `未来${Math.abs(relativeTime[0])}-${Math.abs(relativeTime[1])}天内`;
          }
          default:
            return opMap[operation] + moment(value[0]).format('YYYY-MM-DD');
        }
      }
    }
    return '选择过滤条件';
  };
  const conditionText = useMemo<string>(() => parseValuesToText(valueType, op, values), [valueType, op, values]);
  const visibleChange = (v: boolean) => {
    setVisible(v);
  };
  const curryDimensionValueRequest = ((timeRangeValue: string, measurementsValue: any[]) => {
    return (dimension: string, keyword: string) => {
      return dimensionValueRequest?.({ dimension, timeRange: timeRangeValue, metrics: measurementsValue, keyword });
    };
  })(timeRange, measurements);

  const submit = (v: FilterValueType) => {
    setVisible(false);
    onSubmit(v);
  };
  const cancel = () => {
    setVisible(false);
    onCancel();
  };
  return exprKey ? (
    <Dropdown
      visible={visible}
      trigger={['click']}
      onVisibleChange={visibleChange}
      overlay={
        <FilterAttrOverlay
          valueType={valueType}
          onSubmit={submit}
          onCancel={cancel}
          op={op}
          curryDimensionValueRequest={curryDimensionValueRequest}
          values={values}
          exprKey={exprKey}
        />
      }
      placement="bottomRight"
      getTooltipContainer={() => document.getElementById('expression-box') || document.body}
      destroyTooltipOnHide
    >
      <span className="filter-condition_select">
        <Tooltip title={conditionText} disabled={conditionText === '选择过滤条件'} placement="topLeft">
          <span className="filter-condition_select-text">{conditionText}</span>
        </Tooltip>
        <DownFilled
          size="14px"
          style={{
            transform: visible ? 'rotate(0.5turn)' : '',
            position: 'absolute',
            right: '4px',
            top: '7px',
          }}
        />
      </span>
    </Dropdown>
  ) : null;
}
export default FilterCondition;
