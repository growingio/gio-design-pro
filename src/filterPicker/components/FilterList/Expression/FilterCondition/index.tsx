import React, { useMemo, useState } from 'react';
import { DownFilled } from '@gio-design/icons';
import { Dropdown, Tooltip } from '@gio-design-new/components';
import moment from 'moment';

import FilterAttrOverlay from './FilterAttrOverlay';
import { attributeValue, FilterValueType, StringValue, NumberValue, DateValue } from './interfaces';
// import { selectValueMap } from '../../../../filterMap';

interface FilterConditionProps {
  valueType: attributeValue;
  onSubmit: (v: FilterValueType) => void;
  op: StringValue | NumberValue | DateValue;
  dimensionValueRequest?: (data: any) => Promise<any>;
  timeRange: string;
  measurements: any[];
  values: string[];
  exprKey: string;
}

const operationMap = {
  string: {
    '=': '等于',
    '!=': '不等于',
    in: '在,范围内',
    'not in': '不在,范围内',
    like: '包含',
    'not like': '不包含',
    hasValue: '有值',
    noValue: '无值',
  },
  int: {
    '=': '等于',
    '!=': '不等于',
    '>': '大于',
    '>=': '大于等于',
    '<': '小于',
    '<=': '小于等于',
    between: '在,与,之间',
    hasValue: '有值',
  },
  date: {
    '=': '等于',
    '!=': '不等于',
    '>': '在,天之后',
    '>=': '在,天之后包括当天',
    '<': '在,天之前',
    '<=': '在,天之前包括当天',
    between: '在,与,之间',
    relativeCurrent: '相对现在',
    relativeBetween: '相对区间',
    hasValue: '有值',
  },
};

function FilterCondition(props: FilterConditionProps) {
  const { valueType = 'string', onSubmit, op, dimensionValueRequest, timeRange, measurements, values, exprKey } = props;
  console.log(valueType, 'valueTYpe');
  const [visible, setVisible] = useState(false);
  const parseValuesToText = (type: string, operation: string, value: string[]): string => {
    console.log('string');
    const opMap = operationMap[type];
    if (value.length) {
      if (type === 'string') {
        switch (operation) {
          case '=': {
            if (value[0] === ' ') {
              return '无值';
            }
            return opMap[operation] + value[0];
          }
          case '!=': {
            if (value[0] === ' ') {
              return '有值';
            }
            return opMap[operation] + value[0];
          }

          case 'in':
          case 'not in': {
            const textList = opMap[operation].split(',');
            return textList[0] + value.join(',') + textList[1];
          }
          default:
            return opMap[operation] + value[0];
        }
      }
      if (type === 'int') {
        switch (operation) {
          case '!=': {
            if (value[0] === ' ') {
              return '有值';
            }
            return opMap[operation] + value[0];
          }
          case 'between': {
            const textList = opMap[operation].split(',');
            return textList[0] + value[0] + textList[1] + value[1] + textList[2];
          }
          default:
            return opMap[operation] + value[0];
        }
      }
      if (type === 'date') {
        switch (operation) {
          case '!=': {
            if (value[0] === ' ') {
              return '有值';
            }
            return opMap[operation] + moment(value[0]).format('YYYY-MM-DD');
          }
          case '>':
          case '>=':
          case '<':
          case '<=': {
            const textList = opMap[operation].split(',');
            return textList[0] + moment(value[0]).format('YYYY-MM-DD') + textList[1];
          }
          case 'between': {
            const textList = opMap[operation].split(',');
            const abs = value[0].split(':')[1].split(',');
            console.log(abs, 'abs');
            console.log(moment(parseInt(abs[0], 10)).valueOf(), new Date(abs[0]), 'formatTime');
            return (
              textList[0] +
              moment(parseInt(abs[0], 10)).format('YYYY-MM-DD') +
              textList[1] +
              moment(parseInt(abs[1], 10)).format('YYYY-MM-DD') +
              textList[2]
            );
          }
          case 'relativeTime': {
            const relativeTime = value[0]
              .split(':')[1]
              .split(',')
              .map((ele: string) => parseInt(ele, 10));
            if (relativeTime.length === 1) {
              const day = relativeTime[0];
              if (day < 0) {
                return `过去${Math.abs(day)}天前`;
              }
              return `未来${Math.abs(day)}天后`;
            }
            if (relativeTime.includes(0)) {
              if (relativeTime[0]) {
                if (relativeTime[0] < 0) {
                  return `过去${Math.abs(relativeTime[0])}天内`;
                }
                return `未来${Math.abs(relativeTime[0])}天内`;
              }
              if (relativeTime[1] < 0) {
                return `过去${Math.abs(relativeTime[1])}天内`;
              }
              return `未来${Math.abs(relativeTime[1])}天内`;
            }
            if (relativeTime[0] < 0) {
              return `过去${Math.abs(relativeTime[0])}-${Math.abs(relativeTime[1])}天内`;
            }
            return `未来${Math.abs(relativeTime[0])}-${Math.abs(relativeTime[1])}天内`;
          }
          default:
            return opMap[operation] + moment(value[0]).format('YYYY-MM-DD');
        }
      }
    }
    return '选择过滤条件';
  };
  const conditionText = useMemo<string>(() => parseValuesToText(valueType, op, values), [valueType, op, values]);
  const visibleChange = (v: boolean) => {
    setVisible(v);
  };
  const curryDimensionValueRequest = ((timeRangeValue: string, measurementsValue: any[]) => {
    return (dimension: string, keyword: string) => {
      return dimensionValueRequest({ dimension, timeRange: timeRangeValue, metrics: measurementsValue, keyword });
    };
  })(timeRange, measurements);

  // const parseFilterValueToConditionText = (v: filterValueType): string => {
  //   return `${selectValueMap[attribute][v.op]} ${v.values[0]}`;
  // };
  const submit = (v: FilterValueType) => {
    setVisible(false);
    // setConditionText(parseFilterValueToConditionText(v));
    onSubmit(v);
  };
  const cancel = () => {
    setVisible(false);
  };
  return (
    valueType && (
      <Dropdown
        visible={visible}
        trigger={['click']}
        onVisibleChange={visibleChange}
        overlay={
          <FilterAttrOverlay
            valueType={valueType}
            onSubmit={submit}
            onCancel={cancel}
            op={op}
            curryDimensionValueRequest={curryDimensionValueRequest}
            values={values}
            exprKey={exprKey}
          />
        }
        placement="bottomRight"
        getTooltipContainer={() => document.getElementById('expression-box')}
        destroyTooltipOnHide
      >
        <span className="filter-condition_select">
          <Tooltip title={conditionText} disabled={conditionText === '选择过滤条件'} placement="topLeft">
            <span className="filter-condition_select-text">{conditionText}</span>
          </Tooltip>
          <DownFilled
            size="14px"
            style={{
              transform: visible && 'rotate(0.5turn)',
              position: 'absolute',
              right: '4px',
              top: '10px',
            }}
          />
        </span>
      </Dropdown>
    )
  );
}
export default FilterCondition;
