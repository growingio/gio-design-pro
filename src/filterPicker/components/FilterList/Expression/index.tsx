import React, { useEffect, useState } from 'react';
import { DeleteOutlined } from '@gio-design/icons';
import { Button } from '@gio-design/components';
import PropertyPicker from '../../../../property-picker';
import '../../../../property-picker/style/index';
// import PropertySelect from '@gio-core/components/property-select';
import FilterCondition from './FilterCondition';
import './index.less';
import { attributeValue, FilterValueType, StringValue, NumberValue, DateValue } from '../../../interfaces';

interface ExpressionProps {
  index?: number;
  filterItem: FilterValueType;
  deleteFilterItem: (index: number) => void;
  dimensionValueRequest?: (data: any) => Promise<any>;
  timeRange: string;
  measurements: any[];
  onChange: (expression: FilterValueType, index: number) => void;
  propertyOptions: any[];
  exprs: any[];
  recentlyStorePrefix: string;
}
function Expression(props: ExpressionProps) {
  const {
    index = 0,
    filterItem,
    deleteFilterItem,
    dimensionValueRequest,
    timeRange,
    measurements,
    onChange,
    propertyOptions,
    exprs,
    recentlyStorePrefix,
  } = props;
  const [valueType, setValueType] = useState<attributeValue>(filterItem?.valueType || 'string');
  const [values, setValues] = useState<string[]>(filterItem?.values);
  const [exprKey, setExprKey] = useState<string>(filterItem?.key || '');
  const [exprName, setExprName] = useState<string>(filterItem?.name || '');
  const [op, setOp] = useState<StringValue | NumberValue | DateValue>(filterItem?.op);
  const [subFilterItem, setSubFilterItem] = useState<FilterValueType>(filterItem);

  const submit = (v: FilterValueType) => {
    const expr: FilterValueType = {
      key: exprKey,
      name: exprName,
      valueType,
      ...v,
    };
    v && setValues(v.values);
    v && setOp(v.op);
    setSubFilterItem(expr);
    onChange(expr, index);
  };

  useEffect(() => {
    setSubFilterItem(filterItem);
  }, [filterItem]);

  const cancel = () => {
    onChange(subFilterItem, index);
  };

  const changePropertyPicker = (v: any) => {
    v && setValueType(v?.valueType?.toLowerCase() || 'string');
    v && setExprName(v.label);
    v && setExprKey(v.value);
    v && setValues([]);
    v && setOp('=');
    const expr: FilterValueType = {
      key: v.value,
      name: v.label,
      valueType: v?.valueType?.toLowerCase() || 'string',
      op: '=',
      values: [],
    };
    onChange(expr, index);

    // 兼容旧版方法
    // v && setValueType(v?.valueType?.toLowerCase() || 'string');
    // v && setExprName(v.label);
    // v && setExprKey(v.key);
    // v && setValues([]);
    // v && setOp('=');
    // const expr: FilterValueType = {
    //   key: v.key,
    //   name: v.label,
    //   valueType: v?.valueType?.toLowerCase() || 'string',
    //   op: '=',
    //   values: [],
    // };
    // onChange(expr, index);
  };
  // 兼容旧版属性选择器方法
  // const getGroupIcon = () => (
  //   <span className="group-icon">
  //     {/* <Icon name={`gicon-${group}`} /> */}
  //     <TagOutlined />
  //   </span>
  // );
  // // 兼容旧版属性选择器;
  // const defaultStyle = {
  //   width: '180px',
  //   height: '40px',
  //   placeholder: '选择维度',
  //   disabled: false,
  // };

  return (
    <div className="expression-box" id="expression-box">
      <div className="express-regular_select">
        <div className="expression-icon">{index + 1}</div>
        <PropertyPicker
          initialValue={{ value: exprKey, label: exprName }}
          dataSource={propertyOptions.filter((option: any) => {
            const inavailableOptions = exprs ? exprs.map((expr: any) => expr.key) : [];
            return option.id === exprKey || inavailableOptions.indexOf(option.id) === -1; // && !(/like/.test(operator) && option.id === 'cs1'）saas老逻辑，暂时不需要
          })}
          onChange={changePropertyPicker}
          recentlyStorePrefix={recentlyStorePrefix}
        />
        {/* <PropertySelect
          key="dimension-select"
          value={!exprKey || propertyOptions.some((d: any) => d.id === exprKey) ? exprKey : '无效维度'}
          options={propertyOptions.filter((option: any) => {
            const inavailableOptions = exprs ? exprs.map((expr: any) => expr.key) : [];
            return option.id === exprKey || inavailableOptions.indexOf(option.id) === -1; // && !(/like/.test(operator) && option.id === 'cs1'）saas老逻辑，暂时不需要
          })}
          grouped
          showSearch
          dropdownMatchSelectWidth={false}
          style={defaultStyle}
          dropdownStyle={{ width: '360px' }}
          placeholder="选择维度"
          onChange={changePropertyPicker}
          getGroupIcon={getGroupIcon}
          notFoundContent="没有可用维度"
          className="dimension-select dimension-select-dropdown"
        /> */}
        <FilterCondition
          valueType={valueType}
          onSubmit={submit}
          op={op}
          dimensionValueRequest={dimensionValueRequest}
          timeRange={timeRange}
          measurements={measurements}
          values={values}
          exprKey={exprKey}
          onCancel={cancel}
        />
      </div>
      <Button type="assist" icon={<DeleteOutlined />} size="middle" onClick={() => deleteFilterItem(index)} />
    </div>
  );
}

export default Expression;
