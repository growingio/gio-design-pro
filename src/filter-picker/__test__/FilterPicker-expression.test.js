import React from 'react';
import { shallow } from 'enzyme';
import Expression from '../components/FilterList/Expression';

const exprs = [
  {
    op: '=',
    values: [],
    valueType: 'string',
    name: '',
    key: '',
  },
];

const filterItem = exprs[0];

const propertyOptions = [
  {
    id: 'd',
    name: '域名',
    groupId: 'normal',
    groupName: '常用维度',
    type: 'global',
    valueType: null,
    __typename: 'Dimension',
  },
];

const getExpression = () => <Expression propertyOptions={propertyOptions} exprs={exprs} filterItem={filterItem} />;

describe('expression test', () => {
  it('should match Expression snapshot.', () => {
    const wrapper = shallow(getExpression());
    expect(wrapper.render()).toMatchSnapshot();
  });
});
