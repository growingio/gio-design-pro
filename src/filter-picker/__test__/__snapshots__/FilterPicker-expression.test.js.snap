// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expression test should match Expression snapshot. 1`] = `
<div
  class="expression-box"
  id="expression-box"
>
  <div
    class="express-regular_select"
  >
    <div
      class="expression-icon"
    >
      1
    </div>
    <div
      class="gio-selector gio-selector--middle gio-selector--borderless gio-property-selector"
      style="cursor:pointer"
    >
      <span
        class="gio-selector__placeholder"
      >
        选择属性
      </span>
      <span
        aria-label="down-filled"
        class="gio-icon gio-selector__arrow"
        role="img"
      >
        <svg
          fill="currentColor"
          height="14px"
          viewBox="0 0 14 14"
          width="14px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <g>
            <g>
              <path
                d="M7.39 9.51a.5.5 0 01-.78 0l-3-3.7a.51.51 0 01-.06-.53A.5.5 0 014 5h6a.5.5 0 01.45.28.51.51 0 01-.06.53z"
                fill="currentColor"
                fill-rule="evenodd"
                id="down-filled_svg__down-2"
              />
            </g>
          </g>
        </svg>
      </span>
    </div>
  </div>
  <button
    class="gio-btn gio-btn-text gio-btn-middle gio-btn-icon-only"
    type="button"
  >
    <span
      aria-label="delete-outlined"
      class="gio-icon"
      role="img"
    >
      <svg
        fill="currentColor"
        height="14px"
        viewBox="0 0 14 14"
        width="14px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <g>
          <g>
            <path
              d="M9 1.2a.2.2 0 00-.2-.2H5.2a.2.2 0 00-.2.2v.6a.2.2 0 00.2.2h3.6a.2.2 0 00.2-.2zm1 9.3a.5.5 0 01-1 0v-5a.5.5 0 011 0zm-5 0a.5.5 0 01-.85.35.47.47 0 01-.15-.35v-5a.5.5 0 01.85-.35.47.47 0 01.15.35zm2.5 0a.5.5 0 01-1 0v-5a.5.5 0 011 0zm4-7A.5.5 0 0011 3H3a.5.5 0 00-.5.5v9a.5.5 0 00.5.5h8a.5.5 0 00.5-.5zm-1.5-2a.47.47 0 00.15.35.47.47 0 00.35.15H13a.5.5 0 010 1h-.34a.17.17 0 00-.16.16V13a1 1 0 01-1 1h-9a1 1 0 01-1-1V3.16A.17.17 0 001.34 3H1a.5.5 0 010-1h2.5a.47.47 0 00.35-.15A.47.47 0 004 1.5V1a1 1 0 01.29-.71A1 1 0 015 0h4a1 1 0 01.71.29A1 1 0 0110 1z"
              fill="currentColor"
              fill-rule="evenodd"
              id="delete-outlined_svg__delete"
            />
          </g>
        </g>
      </svg>
    </span>
  </button>
</div>
`;
